packages/actor-extract-links-extract-tree/lib/ActorExtractLinksTree.ts

```diff
@@ -0,0 +1,74 @@
+import type {
+  IActionExtractLinks,
+  IActorExtractLinksOutput, IActorExtractLinksArgs,
+} from '@comunica/bus-extract-links';
+import { ActorExtractLinks } from '@comunica/bus-extract-links';
+import type { ILink } from '@comunica/bus-rdf-resolve-hypermedia-links';
+import type { IActorTest } from '@comunica/core';
+import { DataFactory } from 'rdf-data-factory';
+import type * as RDF from 'rdf-js';
+
+const DF = new DataFactory<RDF.BaseQuad>();
+
+/**
+ * A comunica Extract Links Tree Extract Links Actor.
+ */
+export class ActorExtractLinksTree extends ActorExtractLinks {
+  public static readonly aNodeType = DF.namedNode('https://w3id.org/tree#node');
+  public static readonly aRelation = DF.namedNode('https://w3id.org/tree#relation');
+  private static readonly rdfTypeNode = DF.namedNode('http://www.w3.org/1999/02/22-rdf-syntax-ns#type');
+
+  public constructor(args: IActorExtractLinksArgs) {
+    super(args);
+  }
+
+  public async test(action: IActionExtractLinks): Promise<IActorTest> {
+    return true;
+  }
+
+  public async run(action: IActionExtractLinks): Promise<IActorExtractLinksOutput> {
+    return new Promise((resolve, reject) => {
+      const metadata = action.metadata;
+      const rootUrl = action.url;
+      const relationNodeWithCurrentNodeHasSubject: Map<string, boolean> = new Map();
```

Comment:
That's a long name :P
Can we simplify to something like `pageDefinedNodes`?

Can probably also become a `Set`, since we're mapping to `boolean`.